api/src/service/cache2.spec.ts

```diff
@@ -509,8 +515,401 @@ describe("The cache", () => {
       assert.instanceOf(Result.unwrap_err(responseFromCache), NotFound);
     });
   });
+
+  describe("The BusinessEvent parsing", () => {
+    context("project events", async () => {
+      const projectStreamItems: any[] = [
+        {
+          publishers: [],
+          keys: [],
+          data: {
+            json: {
+              type: "project_created",
+              source: "http",
+              publisher: "root",
+              project: {
+                id: "90ffe1958bc5734abc70a2464437fe92",
+                status: "open",
+                displayName: "foo",
+                description: "",
+                assignee: "root",
+                thumbnail: "",
+                projectedBudgets: [],
+                permissions: {},
+                additionalData: {},
+              },
+              time: "2019-07-25T12:03:25.315Z",
+            },
+          },
+          confirmations: 0,
+          blocktime: 1,
+          txid: "1",
+        },
+      ];
+      const addUnknownProperty = items => {
+        items.map(item => {
```

Comment:
use forEach (map shouldn't be used with side effects, and the result of map should always be used)

Reply:
yes you're right. In this case I'm using the return value of map, I should right it like this instead:

`const addUnknownProperty = items => {`
`       return items.map(item => {`
`item.data.json.project.additionalUnknownProperty = true;`
`          return item;`
`        });`
`      };`

Reply:
yes and no :) yes, because it _works_, but no, because the closure still has a side effect on item. Even if you'd ignore the return value of `map`, `item.data.json.project` would still be changed.